src/swap/action.rs

```diff
@@ -82,11 +110,13 @@ where
 {
 }
 
+/// Look for the event `E` in our "Memory".
 #[async_trait::async_trait]
 pub trait CheckMemory<E> {
     async fn check_memory(&self) -> anyhow::Result<Option<E>>;
 }
 
+/// Look for the event `E` by attempting to `Load` it from a database.
```

Comment:
That's the implementation, not the trait. The trait doesn't care where you store it. I have renamed it to `LookUpEvent` as you suggested to avoid confusions.